### PR TITLE
add macro average rouge-l to metrax

### PR DESCRIPTION
**Pull request by jshin1394**

### CODE REVIEW

File: src/metrax/nlp_metrics.py [hunk 1]
```diff
@@ -300,90 +313,70 @@ def compute(self) -> jax.Array:
...
-        total_recall=jnp.array(0, jnp.float32),
-        total_f1=jnp.array(0, jnp.float32),
-        num_examples=jnp.array(0, jnp.float32),
+    raise NotImplementedError(
+        'Subclasses must implement _calculate_instance_scores.'
     )
 
+  @classmethod
+  def _get_common_initial_values(cls) -> dict[str, jax.Array]:
+    """Returns a dictionary of common JAX array attributes initialized to zero."""
```
Comment (jeffcarp) [inline src/metrax/nlp_metrics.py:348]: Nit: I think 'JAX' here is redundant (since this is a JAX metrics library). Would `common metric attributes` be clearer?

File: src/metrax/nlp_metrics.py [hunk 2]
```diff
@@ -394,72 +387,227 @@ def from_model_output(
...
+
+
+@flax.struct.dataclass
+class RougeN(RougeBase):
+  r"""Computes macro-averaged ROUGE-N recall, precision, and F1-score.
+
+  This metric first calculates ROUGE-N precision, recall, and F1-score for each
+  individual prediction compared against its single corresponding reference.
+  ROUGE-N
+  scores are based on the number of overlapping n-grams (sequences of n words)
```
Comment (jeffcarp) [inline src/metrax/nlp_metrics.py:530]: Nit: remove extra newline on L529

**Review by jeffcarp — APPROVED** [review]
Thanks!

File: src/metrax/nlp_metrics.py (hunk 1 above)
Comment (jshin1394) [reply src/metrax/nlp_metrics.py:348]: modified accordingly.

File: src/metrax/nlp_metrics.py (hunk 2 above)
Comment (jshin1394) [reply src/metrax/nlp_metrics.py]: removed.
